src/pages/contributing.md

```diff
@@ -1,43 +1,200 @@
+
+public float applyDiscount(float price) {
+    float newPrice = price;
+    // Apply discount to newPrice
+    // ...
+    return newPrice;
+}
+```
+
+If the pseudocode is not explanatory enough to understand the concept, then a full code example can be used instead. A full code example should supply dummy values and explain what they represent.
+
+## Minor and Patch Changes
+
+If a change occurs between a minor or patch versions of NeoForge, then relevant changes in the documentation should be split into separate sections or put into tabs. This maintains the accuracy of the information depending on the version the modder is currently developing for.
+
+## Style Guide
+
+This documentation uses [Docusaurus][docusaurus], which internally uses [MDX][mdx], to generate the pages. You can find more detailed information about available features on their pages. This style guide will be more focused towards common features and formatting we use in the Markdown files.
+
+### Front Matter
+
+Front matter defines metadata fields which can affect how the page is rendered. This is denoted using `---`, similar to a code block. The only front matter that should usually be defined is `sidebar_position`, which determines where the page should be rendered on the sidebar.
+
+There are other metadata fields like `title` and `description`, but those are typically parsed from the page itself.
+
+```md
+<!-- Second page in the sidebar -->
+---
+sidebar_position: 2
+---
+```
+
+#### Categories
+
+Categories are folders within the documentation. They inherit titles and positional data from `index.md`. If an `index.md` is not within a subfolder, a `_category_.json` file should be created, specifying the `label` representing the name of the section, and `position` representing where on the sidebar it should go.
+
+```js
+{
+    // Name of the category to display
+    "label": "Example Title",
+
+    // This will be rendered as the third element on the sidebar at the current level.
+    "position": 3
+}
+```
 
-Titles should be capitalized in the standard titling format. For example,
+### Titles
 
-* Guide For Contributing to This Documentation
-* Building and Testing Your Mod
+Titles are defined using up to six hashtags (`#`) to define each section. Titles should capitalize everything but unimportant words.
 
-Essentially, capitalize everything but unimportant words.
+```md
+<!-- There should only be one title with a single '#'. -->
+# Guide For Contributing to This Documentation
 
-Spelling, grammar, and syntax should follow those of American English. Also, prefer using separate words over contractions (e.g. "are not" instead of "aren't").
+### Building and Testing Your Mod
+```
+
+### Diction
+
+Spelling, grammar, and syntax should follow those in American English. Avoid using contractions in sentences; use two separate words ('is not' instead of 'isn't'). Additionally, avoid using pronouns when possible, unless you need to directly refer to the reader.
```

Comment:
As an addendum to the last sentence, perhaps it should include a brief reason on why to avoid pronouns? That being, to avoid ambiguity as to what the pronoun refers to (i.e. its antecedent).